build_tools/azure/update_environments_and_lock_files.py

```diff
@@ -0,0 +1,406 @@
+"""Script to update CI environment files and associated lock files.
+
+To run it you need to be in the root folder of the scikit-learn repo:
+python build_tools/azure/update_environments_and_lock_files.py
```

Comment:
May we add a comment about how we need to install `conda-lock==1.0.3` to run this script?

Reply:
I have added a comment saying that the script needs conda-lock and that the version should match the one used in the CI in `build_tools/azure/install.sh`

Reply:
Side-comment: something that still can be done is to only install conda-lock in the CI for lock files that need it (i.e. that uses a mix of pip and conda). The explicit lock files can be used with `conda install` directly.